File: examples/python/config/hooks/pre-tool-use.py
#!/usr/bin/env python3
"""
Pre-tool-use Hook示例

在工具执行前运行，可用于：
- 权限检查
- 参数验证
- 审计日志
- 动态审批
"""

def pre_tool_use(tool_name: str, tool_input: dict) -> dict:
    raise NotImplementedError("AI: 实现 pre_tool_use")

File: examples/nodejs/config/hooks/pre-tool-use.js
/**
 * Pre-tool-use Hook示例
 * 
 * 在工具执行前运行，可用于：
 * - 权限检查
 * - 参数验证
 * - 审计日志
 * - 动态审批
 */

function preToolUse(toolName, toolInput) {
    throw new Error("AI: Pre-tool-use Hook示例 在工具执行前运行，可用于： - 权限检查 - 参数验证 - 审计日志 - 动态审批");
}

module.exports = { preToolUse };
